plugins/hypervisors/xenserver/src/com/cloud/hypervisor/XenServerGuru.java

```diff
@@ -185,7 +185,8 @@ public boolean trackVmHostChange() {
             DataTO srcData = cpyCommand.getSrcTO();
             DataTO destData = cpyCommand.getDestTO();
 
-            if (srcData.getObjectType() == DataObjectType.SNAPSHOT && destData.getObjectType() == DataObjectType.TEMPLATE) {
+            if (srcData.getHypervisorType() == HypervisorType.XenServer && srcData.getObjectType() == DataObjectType.SNAPSHOT &&
```

Comment:
@anshul1886 What you show is implementation dependent. Unless the specification confirms this behaviour, I think .equals is better.